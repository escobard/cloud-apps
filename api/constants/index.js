const dbkeys = require("./dbkeys"),
  routes = require("./routes")

module.exports = {
  dbkeys,
  routes
}
