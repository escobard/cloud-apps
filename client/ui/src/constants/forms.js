export const addNoteFields = [
  {
    name: "subject",
    label: "Subject",
    placeholder: "Enter a subject for your note",
    errors: ["Subject must contain more than 5"]
  },
  {
    name: "note",
    label: "Note *",
    placeholder: "Enter a description for your note (required)",
    errors: ["Note must contain more than 25 characters"]
  }
];
