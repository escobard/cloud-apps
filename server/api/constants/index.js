import dbkeys from "./dbkeys";
import routes from "./routes";

export { dbkeys, routes };
