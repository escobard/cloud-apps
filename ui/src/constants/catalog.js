import { todayDate } from "utils/todayDate";

export const footer = {
  noNotes: "NO NOTES",
  withNotes: "COMPLETED"
};

export const header = {
  title: "Your Notes",
  todayDate: todayDate()
};

export const notes = {
  noNotes: {
    subject: "No Notes",
    note: "Add a note by clicking on the + icon below.",
    icon: {
      image: "exclamation",
      label: "No Notes"
    }
  },
  apiError: {
    subject: "API error",
    icon: {
      image: "exclamation",
      label: "API error"
    }
  }
};
