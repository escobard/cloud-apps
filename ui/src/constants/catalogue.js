import { todayDate } from "utils/todayDate";

export const footer = {
  noNotes: "NO NOTES",
  withNotes: "COMPLETED"
};

export const header = {
  title: "Your Notes",
  todayDate: todayDate()
}