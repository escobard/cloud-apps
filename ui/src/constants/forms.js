export const addNoteFields = [
  {
    name: "subject",
    label: "Subject",
    placeholder: "Enter a subject for your note",
    value: "",
    errors: {
      // expand this later with future error strings
    }
  },
  {
    name: "note",
    label: "Note *",
    placeholder: "Enter a description for your note (required)",
    value: "",
    errors: {}
  }
];