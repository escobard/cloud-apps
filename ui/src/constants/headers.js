export const headers = { "Access-Control-Allow-Origin": "*" };
