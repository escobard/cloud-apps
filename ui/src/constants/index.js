import { addNoteFields } from "./forms";
import { apiRoutes } from "./urls";
import { headers } from "./headers";
import { notes, footer, header } from "./catalog";

export { apiRoutes, addNoteFields, headers, notes, footer, header };
