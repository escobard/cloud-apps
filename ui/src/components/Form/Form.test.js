import DynamicForm from "./index";
import { addNoteFields } from "constants/forms";

describe("Form", () => {
  const testFunct = jest.fn();
  const props = {
    id: "test-id",
    message: {
      title: "test title",
      message: "test message",
      status: false
    },
    addNote: testFunct,
    fields: addNoteFields
  }
  it(">> snapshot is up to date", () => {
    const { container } = render(<DynamicForm {...props} />);
    expect(container).toMatchSnapshot();
  });
});
