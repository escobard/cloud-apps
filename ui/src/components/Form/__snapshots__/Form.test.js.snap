// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form >> snapshot is up to date 1`] = `
<div>
  <form
    class="ui form"
    id="test-id-form"
  >
    <div
      class="field"
    >
      <label
        for="subject"
      >
        Subject
      </label>
      <input
        name="subject"
        placeholder="Enter a subject for your note"
      />
    </div>
    <div
      class="field"
    >
      <label
        for="note"
      >
        Note *
      </label>
      <textarea
        name="note"
        placeholder="Enter a description for your note (required)"
      />
    </div>
    <div
      class="field"
    >
      <button
        class="ui button"
      >
        Add Note
      </button>
    </div>
  </form>
</div>
`;
